### PR TITLE
Objective-C Type Injection

### LINKED ISSUE #15: Inject Type Info via objc_alloc
With the types from #14 we can now start applying them:

We need an analysis that  over all `objc_alloc` callsites (i.e. all P-Code CALL ops with the first argument being the `objc_alloc` address) and then retrieve the constant value behind the second argument. It won't be a const varnode directly but a unique varnode, so you will have to go backwards via the definitions to the const varnode.

Then we need to map the address back to the type somehow. I see a few options for this, not sure which works best. The most straightforward one is probably getting the symbol at that address and looking for a matching type name. Later we might want to coordinate this via some class or service that keeps track of those mappings.

Then we need use the same API to override the call signature at an address which is used by `OverridePrototypeAction`

`AnalyzerType.FUNCTION_ANALYZER` with `AnalysisPriority.FUNCTION_ANALYSIS.after()`

### PR DESCRIPTION
**Pull request by AngeloD2022**
Resolves #15

### CODE REVIEW

**Comment by AngeloD2022** [comment]
@fmagin Do we want to merge this or possibly investigate optimizations to this pass?

**Comment by fmagin** [comment]
We need to still fix the issue that we don't abort/crash the entire analysis when encountering an `objc_alloc(param_N)` call, but as soon as that is fixed we can merge it

**Comment by AngeloD2022** [comment]
It seems there are instances where the alloc function call does not register any constant arguments. This seems to mainly be an issue with `_objc_alloc_init()` from my brief exploration.

**Comment by AngeloD2022** [comment]
I did some further investigation, and both alloc and init_alloc seem to have identical metadata by default (prototype, calling convention, etc). My guess is that this maybe requires some correction.

File: src/main/java/lol/fairplay/ghidraapple/analysis/objcclasses/OCTypeInjectorAnalyzer.kt [hunk 1]
```diff
@@ -90,14 +90,30 @@ class OCTypeInjectorAnalyzer : AbstractAnalyzer(NAME, DESCRIPTION, AnalyzerType.
 
             taskMonitor.checkCancelled()
 
-            val allocCalls: List<PcodeOpAST> = decompiled.highFunction.pcodeOps.asSequence()
+            val highFunction = decompiled.highFunction
+            if (highFunction == null) {
+                Msg.error(this, "Failed to inject types: highFunction is null for ${function.name}")
```
Comment (fmagin) [inline src/main/java/lol/fairplay/ghidraapple/analysis/objcclasses/OCTypeInjectorAnalyzer.kt:95]: This kind of failure seems pretty bad and should be reported via the analysis log instead.
If anything has been logged to it, a popup with the messages will show after the analysis is finished

**Review by fmagin — CHANGES_REQUESTED** [review]

**Comment by fmagin** [comment]
> It seems there are instances where the alloc function call does not register any constant arguments.

This might be alleviated if we explicitly set up the prototype for it before?

**Comment by fmagin** [comment]
@AngeloD2022 I squashed my rewritten implementation of the analyzer down to one commit and force pushed to this branch so we keep the associated discussion with it. I moved your other commits with the P-Code viewer to a new branch for you https://github.com/ReverseApple/GhidraApple/tree/AngeloD2022/pcodeViewer

**Comment by AngeloD2022** [comment]
Love it! Thank you! I'll review in a bit.

**Pull request merged**